test/src/kbasesearchengine/test/events/storage/MongoDBStatusEventStorageTest.java

```diff
@@ -176,6 +179,72 @@ public void storeAndGetNoTypeEmptyCodesAllFields() throws Exception {
         assertThat("incorrect error msg", got.getErrorMessage(), is(Optional.absent()));
         assertThat("incorrect error trace", got.getErrorStackTrace(), is(Optional.absent()));
     }
+
+    @Test
+    public void storeAndGetFalseFields() throws Exception {
+        // tests with all possible fields in status event
+        when(clock.instant()).thenReturn(Instant.ofEpochMilli(30000L));
+        final StoredStatusEvent sse = storage.store(StatusEvent.getBuilder(
+                "WS", Instant.ofEpochMilli(10000), StatusEventType.COPY_ACCESS_GROUP)
+                        .withNullableisPublic(false)
+                        .withNullableOverwriteExistingData(false)
+                        .build(),
+                StatusEventProcessingState.UNPROC,
+                set(),
+                "Baldrick");
+
+        assertThat("incorrect access group id", sse.getEvent().isPublic(), is(Optional.of(false)));
+        assertThat("incorrect access group id", sse.getEvent().isOverwriteExistingData(), is(Optional.of(false)));
```

Comment:
Is there a reason this doesn't test get() like the null fields test?

Also I realize that the tests don't actually test that the true fields work either. They store it but never check it as far as I can tell. Should probably add that.

After that i think this is done

Reply:
I added a couple more test methods. So now storeAndGetNoTypeEmptyCodesAllFields() tests non -default values. storeAndGetAllBooleanFields() tests boolean values for both T/F. storeAndGetNullableDefaults() tests for defaults (which is Optional.absent())